app/assets/stylesheets/long-email-truncation.scss

```diff
@@ -0,0 +1,15 @@
+/********
+Note: this is only designed to account for narrow screens where the Ruby
+truncation to 20 characters is insufficient to prevent overlap.
+/********/
+
+#navigation>li {
+  max-width:calc(50vw - 101px - 30px - 76px - 30px);
+  vertical-align:top;
+  overflow:hidden;
+  text-overflow: ellipsis;
+
+  &:focus-within {
+    color:#0b0c0c;
+  }
+}
```

Comment:
As this is designed to work on narrow screens, would it be worth wrapping it in a `@media (max-width: 350px)` [type directive](https://www.w3schools.com/cssref/css3_pr_mediaquery.asp), so that it is only applied when the screen width drops below a certain width (350px in my example).

Reply:
Here, by narrow screens, we don't mean mobile, but screens of > about 770px but < about 920px.  
The media query could be added here to account for > 920px but because the Ruby truncation kicks in at 20 characters it isn't needed.  A string of 20 upper-case W's does not cause issues.  
<770px is dealt with already by the existing mobile styles

Reply:
I have added the media queries to account for all situations.  This makes the CSS quite robust, dealing with mobile view, narrow desktop/tablet and normal/wide displays.  

Do you think the Ruby truncation to 20 characters is still needed?

Reply:
Talked with @Kalagan, who agrees that the Ruby truncation is now redundant.  So I have removed that, and added a new test email address.